asf_levies_model/notebooks/report_scenarios_summary_v2.py

```diff
@@ -0,0 +1,755 @@
+scenarios_summary_chart_2 = scenarios_summary_chart_2[
+    [
+        "Name",
+        "archetype",
+        "scheme_eligible",
+        "scenario",
+        "main_heating_fuel",
+        "electricity_bill",
+        "gas_bill",
+        "dual_fuel_bill",
+    ]
+].sort_values(by=["scenario", "Name"])
+
+# Create new column for bill change from baseline
+baseline = scenarios_summary_chart_2[
+    scenarios_summary_chart_2["scenario"] == "1. Baseline"
+].set_index("Name")["dual_fuel_bill"]
+
+scenarios_summary_chart_2["Bill change from baseline"] = (
+    scenarios_summary_chart_2.apply(
+        lambda row: row["dual_fuel_bill"] - baseline.get(row["Name"], 0),
+        axis=1,
+    )
+)
+
+scenarios_summary_chart_2 = scenarios_summary_chart_2.reset_index(drop=True)
+
+# %%
+# Construct look up table for full archetype and WHD eligible/ineligible size
+archetype_size = ofgem_archetypes_df[:25]
+archetype_size = archetype_size[["AnnualConsumptionProfile", "ArchetypeSize"]]
+
+eligibility_size = ofgem_archetypes_scheme_eligibility()
+whd_eligibility_size = (
+    eligibility_size[["AnnualConsumptionProfile", "WHDEligibleSize"]]
+    .set_index("AnnualConsumptionProfile")
+    .astype("Int64")
+)
+
+whd_eligibility_df = archetype_size.merge(
+    whd_eligibility_size,
+    how="left",
+    left_on="AnnualConsumptionProfile",
+    right_index=True,
+)
+
+whd_eligibility_df.loc[0, "ArchetypeSize"] = 0
+whd_eligibility_df.loc[0, "WHDEligibleSize"] = 0
+
+whd_eligibility_df["WHDIneligibleSize"] = (
+    whd_eligibility_df["ArchetypeSize"] - whd_eligibility_df["WHDEligibleSize"]
+)
+
+# %%
+# Add column with original archetype name
+scenarios_summary_chart_2["AnnualConsumptionProfile"] = scenarios_summary_chart_2[
+    "Name"
+].str.extract(r"(\w+)")
+
+# Merge summary dataframe with look up dataframe on original archetype name
+scenarios_summary_chart_2 = scenarios_summary_chart_2.merge(
+    whd_eligibility_df, on="AnnualConsumptionProfile", how="left"
+)
+
+# Add column for size based on scheme_eligible
+scenarios_summary_chart_2["size"] = scenarios_summary_chart_2.apply(
+    lambda row: (
+        row["WHDEligibleSize"] if row["scheme_eligible"] else row["WHDIneligibleSize"]
+    ),
+    axis=1,
+)
+
+# Drop processing columns
+scenarios_summary_chart_2 = scenarios_summary_chart_2.drop(
+    columns=[
+        "ArchetypeSize",
+        "WHDEligibleSize",
+        "WHDIneligibleSize",
+        "AnnualConsumptionProfile",
+    ]
+)
+
+# %% [markdown]
+# **Electricity to gas unit cost ratios**
+
+# %%
+# Calculate electricity-to-gas cost ratios for both sets of scenarios
+scenarios = ["baseline"] + list(scenario_weights.keys())
+
+scenario_ratios = [
+    elec_tariffs[scenario].calculate_variable_consumption(1)
+    / gas_tariffs[scenario].calculate_variable_consumption(1)
+    for scenario in scenarios
+]
+
+cost_ratio_frame = pd.DataFrame(
+    {"Scenario": scenarios, "Electricity to gas unit cost ratio": scenario_ratios}
+).reset_index(drop=True)
+
```

Comment:
It was interesting to note that WHD doesn't affect the cost ratio because it is levied on customers not consumption. Even though doubling WHD is more punitive to customers with gas and electricity.